### PR TITLE
Feature/clan6 199 remove cogboard init task

### PR DESCRIPTION
**Pull request by GulderBone**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,8 +10,7 @@ Thumbs.db
 ~*
 
 *.jar
-/knotx/conf/jwt.conf
-/mnt/admins.conf
+/mnt/admins.json
```
Comment (szymon-owczarzak) [inline .gitignore]: redundant - covered by line below

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/CredentialsService.kt [hunk 2]
```diff
@@ -9,17 +9,18 @@ import com.cognifide.cogboard.config.utils.JsonUtils.getObjectPositionById
...
 import com.cognifide.cogboard.config.validation.credentials.CredentialsValidator
 import com.cognifide.cogboard.storage.Storage
+import com.cognifide.cogboard.storage.VolumeStorageFactory
 import io.vertx.core.json.JsonArray
 import io.vertx.core.json.JsonObject
 
 class CredentialsService(
     private val storage: Storage,
-    private val config: JsonObject = storage.loadConfig()
+    private val config: JsonObject = VolumeStorageFactory.credentials().loadConfig()
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/CredentialsService.kt]: we could probably
`import com.cognifide.cogboard.storage.VolumeStorageFactory.credentials`
for better readability

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt [hunk 3]
```diff
@@ -9,12 +9,16 @@ import com.cognifide.cogboard.storage.Storage
...
 import io.vertx.core.json.JsonObject
 
-class EndpointsService(private var storage: Storage, private val config: JsonObject = storage.loadConfig()) {
+class EndpointsService(
+    private var storage: Storage,
+    private val config: JsonObject = VolumeStorageFactory.endpoints().loadConfig()
+) {
 
-    fun loadConfig(): JsonObject = storage.loadConfig()
+    fun loadConfig(): JsonObject = VolumeStorageFactory.endpoints().loadConfig()
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt]: similar as my previous comment

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.cognifide.cogboard.config.validation.admins
+
+import com.cognifide.cogboard.config.validation.Validator
+import java.io.File
+
+object AdminsValidator : Validator {
+    override fun validate(config: String): Boolean {
+        if (File("/data/admins.json").exists()) {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/validation/admins/AdminsValidator.kt]: this validation makes no sense now as i will be always created before code reaches this point... Lets validate if at least one admin is defined with password

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt [hunk 5]
```diff
@@ -14,14 +15,16 @@ import io.vertx.reactivex.ext.web.RoutingContext
 
 class LoginHandler : RoutingHandlerFactory {
 
-    var vertx: Vertx? = null
-    var admins: MutableMap<String, String> = mutableMapOf()
+    private var vertx: Vertx? = null
+    private lateinit var config: JsonObject
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt]: move lateinit field so it is last

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package com.cognifide.cogboard.storage
+
+import com.cognifide.cogboard.widget.type.ExampleWidget
+import io.vertx.core.logging.LoggerFactory
+import java.io.File
+
+class StorageInitializer {
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt]: lets rename this to `ConfigFileCreator`

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+    companion object {
+        private val LOGGER = LoggerFactory.getLogger(ExampleWidget::class.java)
+
+        fun createIfDoesNotExist(configPath: String) {
+            val fileNotExists = !File(configPath).exists()
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt]: we don't need this `val` move it directly to `if ()`

File: .gitignore (hunk 1 above)
Comment (GulderBone) [reply .gitignore]: changed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/CredentialsService.kt (hunk 2 above)
Comment (GulderBone) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/CredentialsService.kt]: changed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt (hunk 3 above)
Comment (GulderBone) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt]: changed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt (hunk 5 above)
Comment (GulderBone) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/security/LoginHandler.kt]: changed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt (hunk 6 above)
Comment (GulderBone) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt]: renamed

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt (hunk 7 above)
Comment (GulderBone) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/StorageInitializer.kt]: moved

**Pull request merged**